pkg/rpcclient/unwrap/unwrap.go

```diff
@@ -167,19 +167,64 @@ func SessionIterator(r *result.Invoke, err error) (uuid.UUID, result.Iterator, e
 	if err != nil {
 		return uuid.UUID{}, result.Iterator{}, err
 	}
-	if t := itm.Type(); t != stackitem.InteropT {
-		return uuid.UUID{}, result.Iterator{}, fmt.Errorf("expected InteropInterface, got %s", t)
-	}
-	iter, ok := itm.Value().(result.Iterator)
-	if !ok {
-		return uuid.UUID{}, result.Iterator{}, errors.New("the item is InteropInterface, but not an Iterator")
+	iter, err := itemToSessionIterator(itm)
+	if err != nil {
+		return uuid.UUID{}, result.Iterator{}, err
 	}
 	if (r.Session == uuid.UUID{}) && iter.ID != nil {
 		return uuid.UUID{}, result.Iterator{}, ErrNoSessionID
 	}
 	return r.Session, iter, nil
 }
 
+// ArrayAndSessionIterator expects correct execution (HALT state) with one or two stack
+// items returned. If there is 1 item, it must be an array. If there is a second item,
+// it must be an iterator. This is exactly the result of smartcontract.CreateCallAndPrefetchIteratorScript.
+// NeoGo server SessionsEnabled parameter must be set to true.
```

Comment:
That's an implementation-specific comment.

Reply:
This method requires sessions to be enabled to the server side. We may replace it with implementation-neutral comment, but still, it won't work properly without iterators, is it OK?

Reply:
The behavior is perfectly fine and expected. The comment can be improved.

Reply:
How can it be improved? We explicitly fail here unless Neo RPC supports sessions (and we have `Session` in name)

Reply:
"Sessions must be enabled on the RPC server for this to function correctly"

Smth like that. Sessions are not unique to NeoGo.

Reply:
Fixed.